Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp

```diff
@@ -1369,21 +1369,26 @@ namespace AZ::Data
         {
             AZ_Assert(itr->second->GetContainerAssetId() == assetId,
                 "Asset container is incorrectly associated with the asset being destroyed.");
-            itr->second->ClearRootAsset();
 
-            // Only remove owned asset containers if they aren't currently loading.
-            // If they *are* currently loading, removing them could cause dependent asset loads that were triggered to
-            // remain in a perpetual loading state.  Instead, leave the containers for now, they will get removed during
-            // the OnAssetContainerReady callback.
-            if (!itr->second->IsLoading())
+            // Make sure the asset instance we're releasing is the same as the one for the container
+            // Sometimes old references (from before a reload) are released which should not cancel newer loads
+            if (itr->second->GetRootAsset()->GetCreationToken() == asset->GetCreationToken())
```

Comment:
nit: Theres a lot of unchecked reach through here, which can have two side effects:

* You are coupling behaviour together in a chain. All the items in the chain are linked together in terms of implementation making future changes harder.
* You are creating a line of potential null ptrs failures.

In this case it may be fine, as all the ptrs are guaranteed to be non-null and its fine for AssetManager to know something has a root asset which has a creation token.

Its something to be aware of.

Reply:
Most of it is sure to be available, the CreationToken is just an int so if its not set it'll default to -1 which is fine and expected.  GetRootAsset might come back null... maybe... I can add a check for that.